public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -522,17 +523,36 @@ export class PrometheusDatasource extends DataSourceApi<PromQuery, PromOptions>
   };
 
   async performSuggestQuery(query: string, cache = false) {
-    if (cache && this.metricsNameCache?.expire > Date.now()) {
-      return this.metricsNameCache.data.filter((metricName: any) => metricName.indexOf(query) !== 1);
-    }
+    let range = this.getTimeRange();
```

Comment:
`performSuggestQuery ` is not used anywhere, just in tests. Then it would probably make sense to remove it together with tests.